gui/wxpython/gui_core/goutput.py

```diff
@@ -473,54 +465,37 @@ def OnCmdProgress(self, event):
         self.progressbar.SetValue(event.value)
         event.Skip()
 
-    def CmdProtocolSave(self):
-        """Save list of manually entered commands into a text log file"""
-        if self.cmdFileProtocol is None:
-            return  # it should not happen
-
-        try:
-            with open(self.cmdFileProtocol, "a") as output:
-                cmds = self.cmdPrompt.GetCommands()
-                output.write("\n".join(cmds))
-                if len(cmds) > 0:
-                    output.write("\n")
-        except IOError as e:
-            GError(
-                _("Unable to write file '{filePath}'.\n\nDetails: {error}").format(
-                    filePath=self.cmdFileProtocol, error=e
+    def OnCmdExportHistory(self, event):
+        """Export history of executed commands into a file"""
+        dlg = wx.FileDialog(
+            self,
+            message=_("Save file as..."),
+            defaultFile="grass_cmd_log.txt",
+            wildcard=_("%(txt)s (*.txt)|*.txt|%(files)s (*)|*")
+            % {"txt": _("Text files"), "files": _("Files")},
+            style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+        )
+
+        if dlg.ShowModal() == wx.ID_OK:
+            path = dlg.GetPath()
+            try:
+                with open(path, "w") as output:
+                    cmds = self.cmdPrompt.GetCommands()
```

Comment:
This exports only commands from the session. The idea was to export the whole thing, basically just copy the .wxgui_history file. That would be implemented in GPrompt.